language/bytecode_verifier/bytecode_verifier_tests/src/unit_tests/signature_tests.rs

```diff
@@ -8,7 +8,19 @@ use invalid_mutations::signature::{
 };
 use proptest::{collection::vec, prelude::*};
 use types::vm_error::StatusCode;
-use vm::file_format::CompiledModule;
+use vm::file_format::*;
+
+#[test]
+fn test_reference_of_reference() {
+    use SignatureToken::*;
+
+    let mut m = basic_test_module();
```

Comment:
Since the only usage of this is to create dummy module with a custom signature, perhaps we should expose

`dummy_signature_module(signature: Vec<SignatureToken>) -> CompiledModule` instead? 

It seems easier to understand how to use that API, and it's more consistent with `dummy_procedure_module`.

Reply:
This is also used in tests for the other checkers. See #841. I'll check this function in when landing #841 and rebase this PR onto that one later.

Reply:
I think the usages in the other checkers would also benefit from more structured APIs (and several of them could use the `dummy_signature_module` API).

My concern is that 
- the `basic_test_module` approach is not very descriptive--how does the client know what's in a `basic_test_module` already and what's not
- the client has to write a lot of boilerplate to use `basic_test_module` that could be hidden behind a more structured API.

Reply:
1) I agree. Right now the client will have to follow the protocol/convention and expect the module to be structured in a certain way. Not sure if this is an issue though, as we're protected by the tests. How about making the comment clearer and define what exactly this module should contain?
2) Clients of this function are tests that need raw access to the module to express what is not expressible in the IR. (The ones that are expressible should be moved to functional tests.) I don't think structured APIs will be useful for these tests, as it is a common pattern a test gets a valid module, mutate one or two specific items/indices, intentionally breaking the structure.